### DISCUSSION TITLE
Government should be listed per planet in the map UI

### DISCUSSION THREAD

**Discussion by brendanjones** [endless-sky/endless-sky#7170] [Balance, Mechanics, and Interactions]
You can't tell who owns what planet on the map UI. Government is listed as a system-wide attribute, so in systems with planets owned by more than one government, you can't tell who owns what planet from the map. This is an unnecessary limitation as it's simple to list the government per-planet on the map UI.

it should be clear to players why they have differing levels of access to/reputation with two planets in the same system. 

Any reasons to not do it?

<img width="260" alt="Screen Shot 2022-08-25 at 08 08 08" src="https://user-images.githubusercontent.com/16049594/186588119-ed0f774d-2de6-41b3-9112-b202260923ad.png">

**Comment by Zitchas** [comment]
It is a good idea, although it might be best if it was optional. By this I mean some way to specify how things show up, as I can think of different situations where showing the government on each planet would be best; and the vast majority of situations where just having it on the system is easiest.

**Comment by Zireael07** [comment]
I'd list gov't per system if all planets are owned by it, and only list per planet if there are outliers, such as the example you cite.

**Reply by brendanjones** [reply]
> I'd list gov't per system if all planets are owned by it, and only list per planet if there are outliers, such as the example you cite.

I like this for simplicity’s sake, though it does mean inconsistent placement of info in the UI. @Zitchas a ‘lore’ question: are systems always owned by one species/faction, even if another owns a planet within that system? If so then I’d say always display the system-level ownership, and also display ownership per-planet if more than one government owns a planet in the system.

**Reply by Zitchas** [reply]
I would tentatively say no, and for the lore I would say that were definitely should not be ruling out the possibility of contested systems without a clear "this system is owned by X"

**Reply by brendanjones** [reply]
> for the lore I would say that were definitely should not be ruling out the possibility of contested systems without a clear "this system is owned by X"

that was what I expected, and it makes sense to me.

**Comment by MasterOfGrey** [comment]
I have a simpler proposal that doesn't use up more UI space - simply have the government name for the system and the planet names all be coloured their respective government colours.
It's not the most colour-blind friendly option, but even to those people it should be pretty clear most of the time if one of them stands out different to the others.

**Reply by brendanjones** [reply]
I’m afraid I don’t like this solution because it implies information instead of simply stating it.  I’ve played 1000+ hours of this game and I’ve no idea the official colour of some races. 

Specific reasons not to:
1. It may tell the player  that the planets are owned by different governments, but not which governments. 
2. it could get confused with the reputation colour scheme. 
3. it relies on the player memorising race colours.

**Reply by Zireael07** [reply]
I'd use some sort of symbol/icon instead of colors. Alternately maybe the initial letter of the government - Q for Quarg, etc. - do we have any letters that multiple factions start with?

**Reply by mOctave** [reply]
There are letters that multiple factions start with: all the Korath groups (Kor Efret, Kor Mereti, Kor Sestor, Korath, New factions?), Pug and Pirate, Heliarch and the two kinds of Hai, Remnant and Republic. The symbol/icon would be nice, but it might make it harder for plugin editors who would find themselves suddenly having to update their old plugins to include symbol images for all the factions in the plugins.

**Reply by MasterOfGrey** [reply]
> 1. It may tell the player  that the planets are owned by different governments, but not which governments.
> 2. it could get confused with the reputation colour scheme.
> 3. it relies on the player memorising race colours.

1. Only relevant for the planets that don't align to the government of the system, which is explicitly stated.
2. I would deem this unlikely, since that's a very specific colour scheme.
3. Those colours are also available on the map right in front of your eyes adjacent to these displays, if you click on anything that shows the map by faction breakdown. So no memorising is required, merely correlation with other things on your screen.

**Reply by mOctave** [reply]
Are you suggesting white text for most planets, coloured for planets that don't match, or coloured text for all planets?

**Reply by MasterOfGrey** [reply]
Coloured text for all planets I think.

**Reply by mOctave** [reply]
In that case it would be hard to tell the difference between Hai and Unfettered Hai: I already have a bit of trouble with that on the map screen, I'm willing to bet that people with certain types of colour blindness would have even more trouble.

**Reply by Zireael07** [reply]
@mOctave have you opened an issue for that?

**Reply by mOctave** [reply]
No, I haven't, although that might be a good idea.

**Reply by brendanjones** [reply]
Please don't do this. In addition to my points above, colouring just adds unnecessary colour to the map. It's visual complexity without actually solving for the problem for which I opened this discussion.

**Reply by MasterOfGrey** [reply]
The map is already coloured with these colours though?

**Reply by brendanjones** [reply]
Holy crap, there's a map view I didn't know about. Your comment had me confused because I'd never seen these government colours on the map, so I went to the map and clicked on just everything ... and it seems the only way to get the government colours is to click on "(not yet visited)" or "(has been visited)" on a planet? This is a problem in itself - why would any click on those?! The thought had never crossed my mind, they're just background info about a planet. What has whether you've visited a planet or not got to do with which government owns it? That view should be made way more discoverable, and unattached from whether you've visited the planet.

**Reply by brendanjones** [reply]
There are also multiple flaws with the colours being way too close together:

1. Hai and Unfettered Hai are the same colour. Any government with an individual reputation system should have its own colour (like Free Worlds and Syndicate are human but not the same colour as Republic).
2. Hai, Unfettered Hai, Kor Sestor, Kor Efret, Remnant and Coalition are all way too close in colour.
3. Republic and Korath are too close in colour.
4. Wanderer and Free Worlds are too close in colour (though those are so far apart on the map it's not really an issue).
5. Heliarch is too close to Quarg in colour (though maybe that's intentional, since they're based on a Quarg ringworld?).

I guess having a colour close to another race doesn't matter so much when they're on the opposite side of the map to each other. But there are so many more colours to use so I don't understand why 6 races have to be shades of pink? Certainly Hai should be a completely different colour to Kor Efret/Sestor, that's just confusing given they're next to each other. 

If you use the light/dark versions of each main colour then you don't run the risk of any being confused with other colours. For example Wanderer is lime green, which makes it too close to Quarg yellow. If Wanderers were made light green and Free Worlds dark green, they'd be easier distinguished from each other and Wanderers from Quarg.

**Reply by MasterOfGrey** [reply]
> Holy crap, there's a map view I didn't know about. Your comment had me confused because I'd never seen these government colours on the map, so I went to the map and clicked on just everything ... and it seems the only way to get the government colours is to click on "(not yet visited)" or "(has been visited)" on a planet? This is a problem in itself - why would any click on those?! The thought had never crossed my mind, they're just background info about a planet. What has whether you've visited a planet or not got to do with which government owns it? That view should be made way more discoverable, and unattached from whether you've visited the planet.

I'm fair sure there's another way or two, cause I've been using it since about my second hour playing the game. (I think there's a menu on the right?) I'm very glad you've discovered it though.

> If you use the light/dark versions of each main colour then you don't run the risk of any being confused with other colours.

In light of the above, this is probably a very reasonable approach to make changes in. I would back this.

**Reply by mOctave** [reply]
> it seems the only way to get the government colours is to click on "(not yet visited)" or "(has been visited)" on a planet?

You can also click on the government of the system in "Ports" view. In fact, when I click on the "not yet visited"/"has been visited" text I get a different view about numbers of planets visited. This seems to be either a bug or a misunderstanding.

> Hai and Unfettered Hai are the same colour.

As I mentioned above, they are not _quite_ the same colour (Hai is slightly redder, Unfettered is slightly bluer), but they are probably too close (see https://github.com/endless-sky/endless-sky/issues/7186).

**Reply by brendanjones** [reply]
> You can also click on the government of the system in "Ports" view.

Huh, nothing happens when I click on that. That would indeed be a more logical place to click to unearth the governments view. I presume that's where the link is supposed to be? If so then I need to open a bug report ...

**Reply by quyykk** [reply]
> Huh, nothing happens when I click on that

That's because you are presumably on a fairly recent commit. This is a known bug fixed by #7072.

**Reply by brendanjones** [reply]
To fix up those colours I'd do something like this. @mOctave is that discussion happening somewhere else? Seems like there's more to be done that just what's in #7186 .


Coalition | #0693e3 | Light blue
Free Worlds | #087c42 | Dark green
Hai | #512da8 | Dark purple
Hai Unfettered | #9575cd | Light purple
Heliarch | #ff699b | Light pink
Korath | #880035 | Dark pink
Kor Efret | #0097a7 | Teal
Kor Mereti | #ffc107 | Light orange
Kor Sestor | #e65100 | Dark orange
Pirate | #ff0000 | Red
Pug | #FBFAF0 | White/cream
Quarg | #ffe500 | Yellow
Remnant | #c2185b | Middle pink
Republic | #ff9800 | Middle orange
Syndicate | #000080 | Dark blue (that's 000080, GH is reading it as an issue link)
Wanderer | #4caf50 | Light green

**Reply by mOctave** [reply]
@brendanjones No, I only opened that issue to deal with the Hai/Unfettered colour similarity.

**Reply by mOctave** [reply]
@brendanjones Although serious changes to the colours would help with accessibility, I don't think they would be a good idea overall for a few reasons:

- Several of the colours are very different from the colours that those governments currently use. This means that anyone who has been playing the game using the government map for a long time will be very confused when the Coalition, Free Worlds, Hai, Kor Mereti, and Syndicate undergo serious changes in colour.
- Plugin authors have chosen colours for new factions to fit the current colour scheme. Large changes could cause conflicts with  plugin faction colours.
- Right now, each species' faction has a specific look: humans are bold and vibrant, Coalition and Heliarch systems have a pinkish colour that goes well with the colour of their ships, Korath systems are all fairly dark-coloured, etc. This new colour scheme would break that for all species except the Pug, Wanderers, and Hai.
- Some of the darker colours you suggest would blend in with the background. Conversely, some of the brighter colours (like Pirate), are more garish than those around them. This isn't a major issue, but it would need some work.

Also, a couple of things I'd like to point out that aren't objections but should probably be considered:

- You don't have the Drak included in your list. They don't control any systems (at least not in 0.9.14), but they do use a white colour, which should be accounted for, just in case plugins or future versions of ES give the Drak control over systems. Also, if contested systems are introduced (like Zitchas mentioned in response to Zireal's comment above), there might end up being a new colour for contested systems which would also have to be taken into account.
- In general, players who use the government-based colours will start to memorize them, encountering only a few at a time. The factions with similar colours are generally spaced far enough apart that it isn't a big issue, and most players who find those factions know more or less where they are anyways. The Hai are unlikely to control a system near Remnant space, therefore it's probably okay that the colours are similar.
- Eventually, we're going to run out of obviously different colours as we fill the galaxy with new factions. At that point, keeping colours unique will become harder and harder to do while still having the colours look good.

Overall, I think it would be a better idea to keep the colours fairly similar to what they are already, but some colour changes (Hai and Unfettered, Kor Efret and Kor Mereti, maybe a few others) could be helpful. I'd love to hear some other opinions on this, though, as this might just be me wanting everything to stay the same forever.

**Comment by TomGoodIdea** [comment]
#8938
